### PR TITLE
docs: update the documents with providers' dependencies and docker versions

### PR DESCRIPTION
**Pull request by sujoy-pal144**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [ ] I have performed preliminary testing using the [test suite](../../apps/api/OsmoX.postman_collection.json) to ensure that any existing features are not impacted and any new features are working as expected.
- [x] I have updated the required [api docs](../../apps/api/docs/) as applicable.
- [ ] I have added/updated test cases to the [test suite](../../apps/api/OsmoX.postman_collection.json) as applicable

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login endpoint`)
- [x] Description has been added
- [ ] Related changes have been added (optional)
- [ ] Screenshots have been added (optional)
- [ ] Query request and response examples have been added (as applicable, in case added or updated)
- [x] Documentation changes have been listed (as applicable)
- [ ] Test suite output is added (as applicable)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

---

**Description:**

Updated the documents with dependencies (packages/version they are using or depending on) of the providers.

**Related changes:**

N/A

**Screenshots:**

N/A

**Query request and response:**

N/A

**Documentation changes:**

Updated the documents (mailgun.md, sms-kapsystem.md, sms-Plivo.md, sms-Twilio.md, smtp.md, wa-360Dialog.md, wa-Twilio-Business.md, wa-Twilio.md ) under channels with the dependencies of the providers.

**Test suite output:**

N/A

**Pending actions:**

N/A

**Additional notes:**

N/A

## Summary by CodeRabbit

- **Documentation**
  - Added dependency sections to multiple API channel documentation files including Mailgun, Plivo, Twilio, Kapsystem, SMTP, 360Dialog, and Twilio Business.
  - Updated development and production setup documentation to include specific Docker and Docker Compose version requirements.
  - Updated portal README to specify Docker and Docker Compose version requirements.

### CODE REVIEW

File: apps/api/docs/channels/wa-360Dialog.md [hunk 1]
```diff
@@ -95,6 +95,13 @@ Here's a sample request body:
 }
 ```
 
+### Dependencies
+
+| Package Name       | Version    | Description                                                                                                                                                |
+| --------------- | ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| @nestjs/axios      | ^3.0.2  | Integration of Axios with NestJS for making HTTP requests.     |
+
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/channels/wa-360Dialog.md:103]: Single blank line looks fine, doesn't need double blank lines.

File: apps/api/docs/channels/wa-360Dialog.md [hunk 2]
```diff
@@ -95,6 +95,13 @@
...
 References
 - https://docs.360dialog.com/partner/first-steps/sandbox#example-request-payload
-- https://docs.360dialog.com/partner/first-steps/sandbox#5.-send-a-template-message-optional
+- https://docs.360dialog.com/partner/first-steps/sandbox#5.-send-a-template-message-optional
+
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/channels/wa-360Dialog.md:107]: I haven't done any change in this section for this commit, keeping it as it is.

File: apps/api/docs/channels/sms-Plivo.md [hunk 3]
```diff
@@ -45,6 +45,12 @@ Here's a sample request body:
 }
 ```
 
+### Dependencies
+
+| Package Name       | Version    | Description                                                                                                                                                |
+| --------------- | ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| plivo      | ^4.62.0  | Plivo client for sending SMS and making voice calls.     |
+
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/channels/sms-Plivo.md:53]: I haven't done any change in this section for this commit, keeping it as it is.

File: apps/api/docs/channels/wa-Twilio.md [hunk 4]
```diff
@@ -44,5 +44,12 @@ Here's a sample request body:
...
+### Dependencies
+
+| Package Name       | Version    | Description                                                                                                                                                |
+| --------------- | ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| twilio      | ^5.0.4  | Twilio client for sending SMS, making voice calls, and other communication functionalities.     |
+
 References
-- https://www.twilio.com/docs/whatsapp
\ No newline at end of file
+- https://www.twilio.com/docs/whatsapp
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/channels/wa-Twilio.md:54]: Done, I have made the change.

File: apps/api/docs/channels/sms-Twilio.md [hunk 5]
```diff
@@ -45,6 +45,12 @@ Here's a sample request body:
 }
 ```
 
+### Dependencies
+
+| Package Name       | Version    | Description                                                                                                                                                |
+| --------------- | ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| twilio      | ^5.0.4  | Twilio client for sending SMS, making voice calls, and other communication functionalities.     |
+
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/channels/sms-Twilio.md:53]: I haven't done any change in this section for this commit, keeping it as it is.

File: apps/api/docs/channels/mailgun.md [hunk 6]
```diff
@@ -65,4 +65,11 @@ In addition to the provided fields, there are several other options you can incl
...
-Reference: https://documentation.mailgun.com/en/latest/api-sending.html
\ No newline at end of file
+### Dependencies
+
+| Package Name       | Version    | Description                                                                                                                                                |
+| --------------- | ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| mailgun.js       | ^10.2.1   | A Mailgun client for JavaScript, used to interact with the Mailgun API.     |
+| form-data           | ^4.0.0 | Used by mailgun.js to handle form data for HTTP requests.     | 
+
+Reference: https://documentation.mailgun.com/en/latest/api-sending.html
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/channels/mailgun.md:75]: I haven't done any change in this section for this commit, keeping it as it is.

File: apps/api/docs/channels/smtp.md [hunk 7]
```diff
@@ -71,4 +71,10 @@ In addition to the provided fields, there are several other options you can incl
...
 
-Reference: https://nodemailer.com/message/
\ No newline at end of file
+### Dependencies
+
+| Package Name       | Version    | Description                                                                                                                                                |
+| --------------- | ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| nodemailer      | ^6.9.13  | Nodemailer library for sending emails using SMTP, SMTP pool, sendmail, or direct transport.     |
+
+Reference: https://nodemailer.com/message/
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/channels/smtp.md:80]: I haven't done any change in this section for this commit, keeping it as it is.

File: apps/api/docs/channels/wa-Twilio.md [hunk 8]
```diff
@@ -44,5 +44,13 @@ Here's a sample request body:
 }
 ```
 
+### Dependencies
+
+| Package Name       | Version    | Description                                                                                                                                                |
+| --------------- | ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| twilio      | ^5.0.4  | Twilio client for sending SMS, making voice calls, and other communication functionalities.     |
+
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/channels/wa-Twilio.md:52]: Single blank line looks fine, doesn't need double blank lines.

File: apps/api/docs/channels/wa-Twilio.md [hunk 9]
```diff
@@ -44,5 +44,13 @@
...
 References
-- https://www.twilio.com/docs/whatsapp
+
+- https://www.twilio.com/docs/whatsapp
+
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/channels/wa-Twilio.md:56]: I haven't done any change in this section for this commit, keeping it as it is.

**Comment by LakshayaT** [comment]
@sujoy-pal144 Can we also include docker and docker-compose version details in the docs?

Can add the following versions:
```sh
$ docker --version
Docker version 26.1.2, build 211e74b
$ docker-compose --version
Docker Compose version v2.1.1
```

File: apps/api/docs/production-setup.md [hunk 10]
```diff
@@ -12,6 +12,8 @@ Before setting up OsmoX for production, ensure you have the following prerequisi
 - **MariaDB:** MariaDB v10.x or higher.
 - **Redis:** Redis v6.x or higher
 - **PM2 (Process Manager):** PM2 v5.x or higher.
+- **Docker:** 26.1.2, build 211e74b
+- **Docker Compose:** v2.1.1
 
 These prerequisites are essential for deploying and running OsmoX in a environment.
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/production-setup.md]: Done, updated the sentence.

File: apps/api/docs/development-setup.md [hunk 11]
```diff
@@ -11,6 +11,8 @@ Before setting up OsmoX for development, ensure you have the following prerequis
 - **Git:** Git v2.x or higher.
 - **MariaDB:** MariaDB v10.x or higher.
 - **Redis:** Redis v6.x or higher
+- **Docker:** 26.1.2, build 211e74b
```
Comment (LakshayaT) [inline apps/api/docs/development-setup.md:14]: Do we have docker setup instructions even in development setup document? If not, we can remove from this doc.

Also can preferably change the `**Docker:** 26.1.2, build 211e74b` line to `**Docker:** Docker v26.1.2`. Don't need to specify the build, version should be enough.
Let's keep consistency with other pre-requisites mentioned here too: `**Docker Compose:** v2.1.1`->`**Docker Compose:** Docker Compose v2.1.1`

Also, I believe we have docker setup instructions in the portal README. So please add these details there as well.

**Review by LakshayaT — CHANGES_REQUESTED** [review]

File: apps/api/docs/development-setup.md (hunk 11 above)
Comment (LakshayaT) [reply apps/api/docs/development-setup.md:14]: Can further update the PR title accordingly as we are adding more than just provider dependency versions

File: apps/portal/README.md [hunk 12]
```diff
@@ -39,6 +39,10 @@ Before deploying the Portal with Docker, ensure that you have the following inst
 - [Docker](https://docs.docker.com/get-docker/)
 - [Docker Compose](https://docs.docker.com/compose/install/)
 
+Ensure their versions are:
+- **Docker:** Docker v26.1.2
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/portal/README.md:44]: Done, added the blank lines

File: apps/api/docs/production-setup.md [hunk 13]
```diff
@@ -12,8 +12,10 @@ Before setting up OsmoX for production, ensure you have the following prerequisi
 - **MariaDB:** MariaDB v10.x or higher.
 - **Redis:** Redis v6.x or higher
 - **PM2 (Process Manager):** PM2 v5.x or higher.
+- **Docker:** Docker v26.1.2
+- **Docker Compose:** Docker Compose v2.1.1
```
[the remark this answers is not in the document]
Comment (sujoy-pal144) [reply apps/api/docs/production-setup.md:16]: Not required, it looks fine.

File: apps/api/docs/development-setup.md (hunk 11 above)
Comment (sujoy-pal144) [reply apps/api/docs/development-setup.md]: Done, updated the documents with the above suggestions.

**Review by LakshayaT — APPROVED** [review]

**Pull request merged**